src/app/dataset-view/dataset.component.ts

```diff
@@ -147,33 +160,35 @@ export class DatasetComponent
 
     private initOverviewTab(): void {
         this.datasetViewType = DatasetViewTypeEnum.overview;
-        this.appDatasetService.getDatasetOverview(this.getDatasetId());
+        this.appDatasetService.getDatasetOverview(this.datasetInfo);
     }
 
     private initDataTab(): void {
         this.datasetViewType = DatasetViewTypeEnum.data;
-        this.appDatasetService.getDatasetDataSchema(this.getDatasetId());
+        this.appDatasetService.getDatasetDataSchema(this.datasetInfo);
     }
 
     private initMetadataTab(currentPage: number): void {
-        this.navigationService.navigateToDatasetView(
-            AppValues.defaultUsername,
-            this.getDatasetId(),
-            ProjectLinks.urlDatasetViewMetadataType,
-            currentPage,
-        );
+        if (this.datasetBasics) {
+            this.navigationService.navigateToDatasetView({
```

Comment:
Why are you doing this redirect again here?
We've already removed navigation from these handlers.
They should start calling GraphQL queries immediately.